### PR TITLE
Update the usage of save-output command in GH actions

### PR DESCRIPTION
**Pull request by vasireddy99**
**Description:** 

This PR upgrades the save-output command in GH actions.

**Testing:** 

Local Testing: https://github.com/vasireddy99/aws-otel-lambda/actions/runs/3415194026

Change Reference : https://github.com/vasireddy99/aws-otel-lambda/blob/test/.github/workflows/soaking.yml#L196

**Documentation:** < Describe the documentation added.>

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: .github/workflows/soaking.yml [hunk 1]
```diff
@@ -204,15 +204,15 @@ jobs:
         run: |
           version=$(echo "${{ steps.extract-sdk-layer-arn.outputs.stdout }}" | cut -d : -f 8)
           echo "Found version number: $version"
-          echo "::set-output name=${{ matrix.language }}-${{ matrix.instrumentation-type }}-version::$version"
+          echo ""${{ matrix.language }}-${{ matrix.instrumentation-type }}-version=$version" >> $GITHUB_OUTPUT"
```
Comment (bryan-aguilar) [inline .github/workflows/soaking.yml:207]:
```suggestion
          echo "${{ matrix.language }}-${{ matrix.instrumentation-type }}-version=$version" >> $GITHUB_OUTPUT
```
I believe double quote should only wrap the string and not the redirect. See [this](https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/) blog for reference also.

File: .github/workflows/soaking.yml [hunk 2]
```diff
@@ -204,15 +204,15 @@ jobs:
         run: |
           version=$(echo "${{ steps.extract-sdk-layer-arn.outputs.stdout }}" | cut -d : -f 8)
           echo "Found version number: $version"
-          echo "::set-output name=${{ matrix.language }}-${{ matrix.instrumentation-type }}-version::$version"
+          echo "${{ matrix.language }}-${{ matrix.instrumentation-type }}-version=$version" >> $GITHUB_OUTPUT"
```
Comment (bryan-aguilar) [inline .github/workflows/soaking.yml:207]:
```suggestion
          echo "${{ matrix.language }}-${{ matrix.instrumentation-type }}-version=$version" >> $GITHUB_OUTPUT
```

File: .github/workflows/soaking.yml [hunk 3]
```diff
@@ -204,15 +204,15 @@ jobs:
...
         working-directory: ${{ env.TERRAFORM_DIRECTORY }}
       - name: Set Collector layer version output
         id: set-collector-layer-version-output
         if: ${{ matrix.language == 'dotnet' || matrix.language == 'go' }}
         run: |
           version=$(echo "${{ steps.extract-collector-layer-arn.outputs.stdout }}" | cut -d : -f 8)
           echo "Found version number: $version"
-          echo "::set-output name=${{ matrix.language }}-${{ matrix.instrumentation-type }}-version::$version"
+          echo "${{ matrix.language }}-${{ matrix.instrumentation-type }}-version=$version" >> $GITHUB_OUTPUT"
```
Comment (bryan-aguilar) [inline .github/workflows/soaking.yml:215]:
```suggestion
          echo "${{ matrix.language }}-${{ matrix.instrumentation-type }}-version=$version" >> $GITHUB_OUTPUT
```

File: .github/workflows/soaking.yml [hunk 4]
```diff
@@ -252,7 +252,7 @@ jobs:
       - name: Set output if layer Soak Tests has error
         id: set-layer-if-error-output
         if: ${{ failure() }}
-        run: echo "::set-output name=${{ matrix.language }}-${{ matrix.instrumentation-type }}-error::FAILED"
+        run: echo "${{ matrix.language }}-${{ matrix.instrumentation-type }}-version=$version" >> $GITHUB_OUTPUT"
```
Comment (bryan-aguilar) [inline .github/workflows/soaking.yml:255]:
```suggestion
        run: echo "${{ matrix.language }}-${{ matrix.instrumentation-type }}-version=$version" >> $GITHUB_OUTPUT
```

**Review by PaurushGarg — APPROVED** [review]

**Pull request merged**